text/0340-firehose-l2.md

```diff
@@ -0,0 +1,924 @@
+
+### Logs
+
+Kinesis Data Firehose will send logs to CloudWatch when data transformation or data
+delivery fails.  The CDK will enable logging by default and create a CloudWatch LogGroup
+and LogStream for your Delivery Stream.
+
+You can provide a specific log group to specify where the CDK will create the log streams
+where log events will be sent:
+
+```ts fixture=with-destination
+import * as logs from '@aws-cdk/aws-logs';
+
+const logGroup = new logs.LogGroup(this, 'Log Group');
+new DeliveryStream(this, 'Delivery Stream', {
+  logGroup: logGroup,
+  destinations: [destination],
+});
+```
+
+Logging can also be disabled:
+
+```ts fixture=with-destination
+new DeliveryStream(this, 'Delivery Stream', {
+  loggingEnabled: false,
+  destinations: [destination],
+});
+```
+
+See: [Monitoring using CloudWatch Logs](https://docs.aws.amazon.com/firehose/latest/dev/monitoring-with-cloudwatch-logs.html)
+in the *Kinesis Data Firehose Developer Guide*.
+
+### Metrics
+
+Kinesis Data Firehose sends metrics to CloudWatch so that you can collect and analyze the
+performance of the delivery stream, including data delivery, data ingestion, data
+transformation, format conversion, API usage, encryption, and resource usage. You can then
+use CloudWatch alarms to alert you, for example, when data freshness (the age of the
+oldest record in the delivery stream) exceeds the buffering limit (indicating that data is
+not being delivered to your destination), or when the rate of incoming records exceeds the
+limit of records per second (indicating data is flowing into your delivery stream faster
+than it is configured to process).
+
+CDK provides methods for accessing delivery stream metrics with default configuration,
+such as `metricIncomingBytes`, and `metricIncomingRecords` (see [`IDeliveryStream`](../lib/delivery-stream.ts)
+for a full list). CDK also provides a generic `metric` method that can be used to produce
+metric configurations for any metric provided by Kinesis Data Firehose; the configurations
+are pre-populated with the correct dimensions for the delivery stream.
+
+```ts fixture=with-delivery-stream
+// TODO: confirm this is a valid alarm
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+// Alarm that triggers when the per-second average of incoming bytes exceeds 90% of the current service limit
+const incomingBytesPercentOfLimit = new cloudwatch.MathExpression({
+  expression: 'incomingBytes / 300 / bytePerSecLimit',
+  usingMetrics: {
+    incomingBytes: deliveryStream.metricIncomingBytes({ statistic: cloudwatch.Statistic.SUM }),
+    bytePerSecLimit: deliveryStream.metric('BytesPerSecondLimit'),
+  },
+});
+new Alarm(this, 'Alarm', {
+  metric: incomingBytesPercentOfLimit,
+  threshold: 0.9,
+  evaluationPeriods: 3,
+});
+```
+
+See: [Monitoring Using CloudWatch Metrics](https://docs.aws.amazon.com/firehose/latest/dev/monitoring-with-cloudwatch-metrics.html)
+in the *Kinesis Data Firehose Developer Guide*.
+
+## Compression
```

Comment:
Should this not be part of the S3 section?

Reply:
I don't think so, since it does apply to the other destinations where S3 is merely used as a stepping stone and we do our best to remove any need to know that S3 is used at all